pkg/scheduler/service/service.go

```diff
@@ -547,7 +547,7 @@ func (s *SchedulerServer) bootstrap(hks hyperkube.Interface) (*ha.SchedulerProce
 		ReconcileCooldown: s.ReconcileCooldown,
 	})
 
-	masterUri := kmcloud.MasterURI()
+	masterUri := kmcloud.CloudProvider.MasterURI()
```

Comment:
Good call, thought about this when implementing and then subsequently forgot about it.  I'll change the factory function we supply when registering to be idempotent to avoid initializing multiple times.